src/main/java/org/jenkinsci/plugins/workflow/cps/CpsScmFlowDefinition.java

```diff
@@ -96,10 +101,26 @@ public boolean isLightweight() {
         }
         Run<?,?> build = (Run<?,?>) _build;
         if (isLightweight()) {
-            try (SCMFileSystem fs = SCMFileSystem.of(build.getParent(), scm)) {
+            try (SCMFileSystem fs = SCMFileSystem.of(build.getParent(), scm, SCMRevisionAction.getRevision(build))) {
                 if (fs != null) {
                     String script = fs.child(scriptPath).contentAsString();
                     listener.getLogger().println("Obtained " + scriptPath + " from " + scm.getKey());
+                    // a heavyweight checkout will populate the changelog only if there is a previous build
+                    // to compare with
```

Comment:
That would be true of a lightweight checkout as well. A changelog is only meaningful when there is some previous build.

Reply:
Which is why I added the comment... so that the `if (prev != null)` makes sense... we could choose to say this is the first build, so "oh here's the last 1000 changes"... but that would deviate from the behaviour of a heavyweight checkout